activerecord/CHANGELOG.md

```diff
@@ -573,5 +573,21 @@
 
     *Santiago Perez Perret*
 
+    Filter attributes in SQL logs
+
+    Previously, SQL queries in logs containing `ActiveRecord::Base.filter_attributes` were not filtered.
+
+    Now, the filter attributes will be masked `[FILTERED]` in the logs.
+
+    ```
+    # Before:
+      Foo Load (0.2ms)  SELECT "foos".* FROM "foos" WHERE "foos"."passw" = ? LIMIT ?  [["passw", "hello"], ["LIMIT", 1]]
+
+    # After:
+      Foo Load (0.5ms)  SELECT "foos".* FROM "foos" WHERE "foos"."passw" = ? LIMIT ?  [["passw", "[FILTERED]"], ["LIMIT", 1]]
```

Comment:
If `prepared_statements: false` (which is by default for mysql2 adapter), should the SQL be ```Foo Load (0.5ms)  SELECT `foos`.* FROM `foos` WHERE `foos`.`passw` = [FILTERED] LIMIT 1```?

Reply:
@kamipo: Nice catch.

It does not look like my solution will work when `prepared_statements` is `false` as the payload will have empty [`binds`](https://github.com/rails/rails/blob/main/activerecord/lib/active_record/log_subscriber.rb#L46) in such a case.

I was not able to find an optimal solution to iterate over the [`sql`](https://github.com/rails/rails/blob/main/activerecord/lib/active_record/log_subscriber.rb#L43) raw statement and filter the attributes. And, have consequently updated the Changelog to mention the filtration is done only when `prepared_statements` is enabled.

I will be happy to work on the case when `prepared_statements = false` as a follow-up. Keen to know your thoughts.

Reply:
@kamipo: When you get a chance, could you let me know your thoughts on the above comment? TIA!